res/controllers/common-controller-scripts.js

```diff
@@ -109,560 +108,594 @@ var msecondstominutes = function(msecs) {
-    // if softStart is called without factor defined, reset to default
-    if (factor === undefined) {
-        factor = 1;
-    }
-    // disable on note-off or zero value note/cc, use default increase rate '1'
-    engine.softStart(
-        parseInt(group.substring(8, 9)), ((status & 0xF0) !== 0x80 && value > 0),
-        factor);
-};
+        // "bpm" was changed in 1.10 to reflect the *adjusted* bpm, but I presume it
+        // was supposed to return the tracks bpm (which it did before the change).
+        // "file_bpm" is supposed to return the set BPM of the loaded track of the
+        // channel.
+        let fRateScale = average / engine.getValue(group, "file_bpm");
 
-// bpm - Used for tapping the desired BPM for a deck
-var bpm = function() {
-};
+        // Adjust the rate:
+        fRateScale = (fRateScale - 1.) / engine.getValue(group, "rateRange");
 
-bpm.tapTime = 0.0;
-bpm.previousTapDelta = 0.0;
-bpm.tap = [];   // Tap sample values
-
-/* -------- ------------------------------------------------------
-        bpm.tapButton
-   Purpose: Sets the tempo of the track on a deck by tapping the desired beats,
-            useful for manually synchronizing a track to an external beat.
-            (This only works if the track's detected BPM value is correct.)
-            Call this each time the tap button is pressed.
-   Input:   Mixxx deck to adjust
-   Output:  -
-   -------- ------------------------------------------------------ */
-bpm.tapButton = function(deck) {
-    const now = new Date() / 1000; // Current time in seconds
-    const tapDelta = now - bpm.tapTime;
-    bpm.tapTime = now;
-
-    // assign tapDelta in cases where the button has not been pressed previously
-    if (bpm.tap.length < 1) {
-        bpm.previousTapDelta = tapDelta;
-    }
-    // reset if longer than two seconds between taps
-    if (tapDelta > 2.0) {
-        bpm.tap = [];
-        return;
+        engine.setValue(
+            group, "rate",
+            fRateScale * engine.getValue(group, "rate_dir"));
     }
-    // reject occurrences of accidental double or missed taps
-    // a tap is considered missed when the delta of this press is 80% longer than the previous one
-    // and a tap is considered double when the delta is shorter than 40% of the previous one.
-    // these numbers are just guesses that produced good results in practice
-    if ((tapDelta > bpm.previousTapDelta * 1.8)||(tapDelta < bpm.previousTapDelta * 0.6)) {
-        return;
-    }
-    bpm.previousTapDelta = tapDelta;
-    bpm.tap.push(60 / tapDelta);
-    // Keep the last 8 samples for averaging
-    if (bpm.tap.length > 8) { bpm.tap.shift(); }
-    let sum = 0;
-    for (let i=0; i<bpm.tap.length; i++) {
-        sum += bpm.tap[i];
-    }
-    const average = sum / bpm.tap.length;
+}
+// Add class bpm to the Global JavaScript object
+// @ts-ignore Same identifier for class and instance needed for backward compatibility
+this.bpm = bpm;
```

Comment:
since `bpm` was effectively a singleton previously and now its a class with state, shouldn't this call the constructor of that class instead of just aliasing it?